### ISSUE TITLE
Proposal: Note that VS Code and GoLand have provided “inlay hints” that can avoid naked parameters

### ISSUE THREAD

**Issue by pan93412** [uber-go/guide#166]
Related part: https://github.com/uber-go/guide/blob/master/style.md#avoid-naked-parameters

Both [VS Code (with Go extension)](https://twitter.com/golang/status/1549106172870987776) and [GoLand](https://www.jetbrains.com/help/go/inlay-hints.html) have provided “inlay hints” that can show every parameters name:

<img width="677" alt="image" src="https://user-images.githubusercontent.com/28441561/216416050-4eaf43d7-3c75-4dcd-b133-4780f18e3223.png">

<img width="731" alt="image" src="https://user-images.githubusercontent.com/28441561/216416231-9be0c4e5-295f-48f3-b98b-bc75acc8b6c7.png">

I wonder if we can add a section mentioning this feature to give the newer codebase another choice – no need to write such a hint explicitly? This feature makes us no longer need to write C-style comments anymore, since maintaining a separate comment can be easily out-of-sync with the original parameter name, and writing such a comment is a bit uninteresting for this IDE era, I think.

**Comment by alxn** [comment]
I don't believe these hints work in sourcegraph or other source code viewers ?

**Comment by prashantv** [comment]
One place where the parameter information is useful is code reviews, and unfortunately this sort of overlay isn't available in the major review tools.

**Comment by sywhang** [comment]
These hints also won't work on other editors like vim/neovim/etc. Last time I checked Go developer survey, there's still a nontrivial number of these users (over 20%, IIRC).
